cg/meta/clean/api.py

```diff
@@ -0,0 +1,111 @@
+        self.status_db = status_db
+        self.housekeeper_api = housekeeper_api
+
+    def get_bundle_files(
+        self, bundle_name: str, started_at: datetime, pipeline: Pipeline
+    ) -> List[hk_models.File]:
+        """Get any bundle files for a specific version"""
+        hk_bundle_version: Optional[hk_models.Version] = self.housekeeper_api.version(
+            bundle=bundle_name, date=started_at
+        )
+        if not hk_bundle_version:
+            LOG.warning(
+                f"Version not found for "
+                f"bundle:{bundle_name}; "
+                f"pipeline: {pipeline}; "
+                f"date {started_at}"
+            )
+            return []
+
+        LOG.info(
+            f"Version found for "
+            f"bundle:{bundle_name}; "
+            f"pipeline: {pipeline}; "
+            f"date {started_at}"
+        )
+        return self.housekeeper_api.get_files(
+            bundle=bundle_name, version=hk_bundle_version.id
+        ).all()
+
+    @staticmethod
+    def has_protected_tags(file: hk_models.File, protected_tags_lists: List[List[str]]) -> bool:
+        """Check if a file has any protected tags"""
+
+        LOG.info(f"File {file.full_path} has the tags {file.tags}")
+        version_file_tags: List[str] = HousekeeperAPI.get_tag_names_from_file(file)
+
+        _has_protected_tags: bool = False
+        for protected_tags in protected_tags_lists:
+
+            protected_tags_on_file = set(version_file_tags).intersection(protected_tags)
+            if protected_tags_on_file:
+                if protected_tags_on_file == set(protected_tags):
+                    LOG.debug(
+                        "File %s has the protected tag(s) %s, skipping.",
+                        file.full_path,
+                        protected_tags_on_file,
+                    )
+                    _has_protected_tags = True
+                    break
+
+        if not _has_protected_tags:
+            LOG.info("File %s has no protected tags.", file.full_path)
+        return _has_protected_tags
+
+    def get_unprotected_existing_bundle_files(self, before: datetime) -> Iterator[hk_models.File]:
+        """Returns all existing bundle files from analyses started before 'before' that hos no protected tags"""
+
+        pipeline: Pipeline
+        for pipeline in Pipeline:
+
+            protected_tags_lists = WORKFLOW_PROTECTED_TAGS.get(pipeline)
+            if not protected_tags_lists:
+                LOG.debug("No protected tags defined for %s, skipping", pipeline)
+                continue
+
+            analyses: Query = self.status_db.get_analyses_before_date(
+                pipeline=pipeline, before=before
+            )
+
+            analysis: models.Analysis
+            for analysis in analyses:
+                LOG.info("Cleaning analysis %s", analysis)
+                version_files: List[hk_models.File] = self.get_bundle_files(
```

Comment:
```suggestion
                hk_version_files: List[hk_models.File] = self.get_bundle_files(
```
To specify a bit more what kind of version we are describing

Reply:
renamed